### PR TITLE
add validators for sms_daily_limit

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Compare sms daily count to sms daily limit and behave as we currently do for the overall message limit.

hidden behind the feature flag `FF_SPIKE_SMS_DAILY_LIMIT`.

~**Note** that the waffles endpoint tests will fail until https://github.com/cds-snc/notification-utils/pull/132 is merged~

~**Note** that the waffles endpoint tests will fail until https://github.com/cds-snc/notification-utils/pull/136 is merged~

# Test instructions | Instructions pour tester la modification

really needs other PRs for it all to work together, in particular
- https://github.com/cds-snc/notification-api/pull/1588
- https://github.com/cds-snc/notification-utils/pull/132

# Release Instructions | Instructions pour le déploiement

Releasing without the feature flag won't affect production. Need to set the flag to use the sms limit.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: requirements-app.txt [hunk 1]
```diff
@@ -50,7 +50,8 @@ MarkupSafe==2.0.1
 # REVIEW: v2 is using sha512 instead of sha1 by default (in v1)
 itsdangerous==2.0.1
 
-git+https://github.com/cds-snc/notifier-utils.git@48.1.0#egg=notifications-utils
+# git+https://github.com/cds-snc/notifier-utils.git@48.1.0#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@add-sms-count-to-redis#egg=notifications-utils
```
Comment (sastels) [inline requirements-app.txt:54]: must remember to change this once the new version of utils is merged 😅

File: requirements.txt [hunk 2]
```diff
@@ -52,7 +52,8 @@ MarkupSafe==2.0.1
 # REVIEW: v2 is using sha512 instead of sha1 by default (in v1)
 itsdangerous==2.0.1
 
-git+https://github.com/cds-snc/notifier-utils.git@48.1.0#egg=notifications-utils
+#git+https://github.com/cds-snc/notifier-utils.git@48.1.0#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@add-sms-count-to-redis#egg=notifications-utils
```
Comment (sastels) [inline requirements.txt:56]: must remember to change this once the new version of utils is merged 😅

File: app/notifications/validators.py [hunk 3]
```diff
@@ -135,6 +161,58 @@ def warn_about_daily_message_limit(service, messages_sent):
...
+    cache_expiration = int(timedelta(days=1).total_seconds())
+
+    # Send a warning when reaching 80% of the daily limit
+    if nearing_sms_daily_limit:
+        cache_key = near_sms_daily_limit_cache_key(service.id)
+        if not redis_store.get(cache_key):
+            redis_store.set(cache_key, current_time, ex=cache_expiration)
+            send_notification_to_service_users(
+                service_id=service.id,
+                template_id=current_app.config["NEAR_DAILY_LIMIT_TEMPLATE_ID"],  # TODO have a template for sms too
```
Comment (andrewleith) [inline app/notifications/validators.py:177]: Following [discussion here](https://gcdigital.slack.com/archives/C0418PT1G6L/p1662476908573219), we just need to update the existing template to show both the sms and email limits in the same email.

File: tests/app/dao/test_services_dao.py [hunk 4]
```diff
@@ -1070,6 +1071,43 @@ def test_dao_fetch_todays_total_message_count_returns_0_with_yesterday_messages(
...
+        assert fetch_todays_total_sms_count(uuid.uuid4()) == 0
+
+    def test_returns_0_with_yesterday_messages(self):
+        service = create_service()
+        sms_template = create_template(service=service, template_type=SMS_TYPE)
+        today = datetime.utcnow().date()
+        yesterday = today - timedelta(days=1)
+        save_notification(create_notification(created_at=yesterday, template=sms_template))
+        assert fetch_todays_total_sms_count(service.id) == 0
+
```
Comment (andrewleith) [inline tests/app/dao/test_services_dao.py:1109]: Could/should we add a test that clears out redis and ensures the count gets re-fetched?

**Review by jzbahrai — APPROVED** [review]

**Pull request merged**